internal/utils/utils.go

```diff
@@ -118,6 +120,23 @@ func PrefixToPath(s string) (string, error) {
 	return string(p), nil
 }
 
+func IsJson(s io.ReaderAt) bool {
+	b := ReaderAtAdapter{readerAt: s}
+
+	bi := bufio.NewReader(&b)
+
+	for {
+		r, _, err := bi.ReadRune()
+		if err != nil {
+			return false
+		}
+
+		if !strings.ContainsRune(" \t\n\r", r) {
+			return r == '{'
```

Comment:
This is too weak JSON detection

A valid JSON could be

```json
true
```


```json
false
```


```json
null
```


```json
[]
```


```json
[{}, false]
```

Please consider using json.Unmarshal it's the best say to validate something is a valid JSON

```go
var v any
err := json.Unmarshal(whatever, &v)

return err == nil
```

Reply:
Yes, it's intentional here, only a very weak json detection is needed.
The json file can be very large, and it's too expensive to read in hundreds mb of memory just for a single judgment condition.

Reply:
Then please add a comment about this.

Maybe you could rename the method to detectJSON as it would make sense it might not be perfect